File: src/types/question.ts
export interface Vote {
  likes: number;
  dislikes: number;
}

export interface Tag {
  tag_id: number;
  name: string;
  description: string;
}

export interface Comment {
  comment_id: number;
  user_id: number;
  username: string;
  comment_text: string;
  created_at: string;
}

export interface Answer {
  answer_id: number;
  user_id: number;
  username: string;
  title: string;
  description: string;
  votes: Vote;
  created_at: string;
  updated_at: string;
  comments: Comment[];
}

export interface Question {
  question_id: number;
  user_id: number;
  username: string;
  title: string;
  description: string;
  votes: Vote;
  created_at: string;
  updated_at: string;
  tags: Tag[];
  answers: Answer[];
  answer_count: number;
}

File: models/Question.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class VoteModel(BaseModel):
    likes: int = 0
    dislikes: int = 0

class TagModel(BaseModel):
    tag_id: int
    name: str
    description: Optional[str] = None

class CommentModel(BaseModel):
    comment_id: int
    user_id: int
    username: str
    comment_text: str
    created_at: datetime

class AnswerModel(BaseModel):
    answer_id: int
    user_id: int
    username: str
    title: Optional[str]
    description: str
    votes: VoteModel
    created_at: datetime
    updated_at: datetime
    comments: List[CommentModel]

class QuestionModel(BaseModel):
    question_id: int
    user_id: int
    username: str
    title: str
    description: Optional[str]
    votes: VoteModel
    created_at: datetime
    updated_at: datetime
    tags: List[TagModel]
    answers: List[AnswerModel]
    answer_count: int